liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -170,4 +173,26 @@ private void connect() {
   private void updateRequestPriority(LocationRequest request) {
     REQUEST_PRIORITY.get(priority).update(request);
   }
+
+  private PendingIntent getPendingIntent() {
+    if (context.get() != null) {
+
+      if (Build.VERSION.SDK_INT >= 26) {
```

Comment:
I think we can clean up this logic by adding something like a `LocationIntentProvider`.  This class could be in charge of providing the `PendingIntent` based on the `Build.VERSION.SDK_INT` so this code would look like:

```
Context context = context.get();
if (context != null) {
    SdkChecker checker = new SdkChecker();
    LocationIntentProvider provider = new LocationIntentProvider(context, sdkChecker, locationListeners);
    Intent intent = provider.buildIntent();
    return PendingIntent.getService(context.get(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
}
```

`SdkChecker` would be a simple class for checking `Build.VERSION.SDK_INT >= 26` with a function like `SdkChecker#isOreoOrAbove()` 

The reason behind all this is for testing - so we can "mock" the build version and see which `PendingIntent` the provider returns. 

```
@Test
public void whenApiVersionIsOreo_providerReturnsBroadcastReceiver() {
    SdkChecker checker = mock(SdkChecker.class);
    when(checker.isOreoOrAbove()).thenReturn(true);
    ...
}
```